### ISSUE TITLE
false positive super-init-not-called for subclasses of typing.Protocol

### ISSUE THREAD

**Issue by ITProKyle** [PyCQA/pylint#4790]
### Bug description

Given the example code below, both `MyProtocol.__init__()` and `ExplicitProtocolImplimentation.__init__()` are being flagged with `super-init-not-called`.

```python
from abc import abstractmethod
from typing import Any, Protocol


class MyProtocol(Protocol):
    @abstractmethod
    def __init__(self, foo: Any) -> None:
        raise NotImplementedError


class ExplicitProtocolImplimentation(MyProtocol):
    def __init__(self, foo: Any) -> None:
        print(foo)
```
The results are the same with/without using `@abstractmethod`
and/or replacing `typing.Protocol` with `typing_extensions.Protocol`.

### Command used

```shell
pylint test.py
```

### Pylint output

```shell
test.py:7:4: W0231: __init__ method from base class 'Protocol' is not called (super-init-not-called)
test.py:12:4: W0231: __init__ method from base class 'MyProtocol' is not called (super-init-not-called)
```

### Expected behavior

I would expect that neither of these would be flagged with `super-init-not-called`.

While calling `super().__init__()` here won't cause a runtime error, it also won't do anything. As noted in [PEP 544](https://www.python.org/dev/peps/pep-0544/), _[Protocols cannot be instantiated](https://www.python.org/dev/peps/pep-0544/#subtyping-relationships-with-other-types)_. This is enforced by the removal of the implementation of `MyProtocol.__init__()` at runtime.

I would also expect that trying to use `super().__init__()` in `MyProtocol.__init__()` or `ExplicitProtocolImplimentation.__init__()` would result in a warning since we can deduce that it is unnecessary (e.g. `superfluous-super`) but this could be addressed in a separate issue as a feature request.

### Pylint version

```shell
pylint 2.9.6
astroid 2.6.5
Python 3.9.6 (default, Jul 15 2021, 14:52:44) 
[GCC 9.3.0]
```
